examples/list.rs

```diff
@@ -1,24 +1,70 @@
-use std::{
-    error::Error,
-    io,
-    time::{Duration, Instant},
-};
+use std::{error::Error, io};
 
 use crossterm::{
     event::{self, DisableMouseCapture, EnableMouseCapture, Event, KeyCode, KeyEventKind},
     execute,
     terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
 };
-use ratatui::{prelude::*, widgets::*};
+use ratatui::{prelude::*, style::palette::tailwind, widgets::*};
+
+const TODO_HEADER_BG: Color = tailwind::BLUE.c950;
+const NORMAL_ROW_COLOR: Color = tailwind::SLATE.c950;
+const ALT_ROW_COLOR: Color = tailwind::SLATE.c900;
+const SELECTED_STYLE_FG: Color = tailwind::BLUE.c300;
+const TEXT_COLOR: Color = tailwind::SLATE.c200;
+const COMPLETED_TEXT_COLOR: Color = tailwind::GREEN.c600;
+
+enum Status {
+    Todo,
+    Completed,
+}
+
+struct TodoItem {
+    todo: String,
+    info: String,
+    status: Status,
+}
+
+impl TodoItem {
+    fn styled_list_item(&self, index: usize) -> ListItem {
+        let bg_color = match index % 2 {
+            0 => NORMAL_ROW_COLOR,
+            _ => ALT_ROW_COLOR,
+        };
+        let (style, todo_str) = match self.status {
+            Status::Completed => (
+                Style::default().fg(COMPLETED_TEXT_COLOR).bg(bg_color),
+                " ✓ ".to_string() + &self.todo,
+            ),
+            Status::Todo => (
+                Style::default().fg(TEXT_COLOR).bg(bg_color),
+                " ☐ ".to_string() + &self.todo,
+            ),
+        };
+        ListItem::new(Line::from(todo_str)).style(style)
+    }
+
+    fn from_vec(vec: Vec<(&str, &str, Status)>) -> Vec<Self> {
+        let mut result_vec = vec![];
+        for item in vec {
+            result_vec.push(TodoItem {
+                todo: item.0.to_string(),
+                info: item.1.to_string(),
+                status: item.2,
+            })
+        }
+        result_vec
+    }
```

Comment:
It wouldn't hurt to move this to the StatefulList and just accept the vec there (it only actually has to be &[])


```rust
impl From<(&str, &str, bool)> for TodoItem {
    fn from((todo, item, status): (&str, &str, bool)) -> Self {
        Self { todo, item, status }
    }
}

impl StatefulList
    fn with_items(items: &[(&str, &str, bool)>]) -> Self {
        Self {
            items: items.map(TodoItem::from).collect()
            state: ListState::default(),
            last_selected: None,
        }
    }
```